src/scraper.rs

```diff
@@ -77,8 +77,8 @@ impl Scraper {
     }
 
     /// Push a new URL into the channel
-    fn push(transmitter: &Sender<(Url, i32)>, url: Url, depth: i32) {
-        if let Err(e) = transmitter.send((url, depth)) {
+    fn push(transmitter: &Sender<(Url, i32, i32)>, url: Url, depth: i32, edepth: i32) {
+        if let Err(e) = transmitter.send((url, depth, edepth)) {
```

Comment:
We will need to change ``edepth`` to whatever name we will chose on the other discussion.